wrapanapi/systems/redfish.py

```diff
@@ -178,7 +311,57 @@ def get_server(self, resource_id):
         """
         return RedfishServer(self, raw=self.api_client.find(resource_id))
 
+    def get_chassis(self, resource_id, *required_types):
+        """
+        Fetch a RedfishChassis instance of the physical chassis representing resource_id.
+
+        In Redfish, a Chassis may be of specific type such as Rack, Sled, Block
+        or any other type. Use the required_types optional parameter to filter
+        by type, or use any of the specific getters such as get_rack.
+
+        Args:
+          resource_id: the Redfish @odata.id of the resource representing the
+             chassis to be retrieved
+          required_types: optional list of ChassisType values that the retrieved
+             chassis needs to belong to
+        Raises:
+           InvalidValueException if the resource_id represents a Chassis that is
+              not of any of the required types
+        """
+        chassis = RedfishChassis(self, raw=self.api_client.find(resource_id))
```

Comment:
Out of curiosity: how does `self.api_client` know that we're after chassis and not some other entity? Is entity type part of ID itself?

Also: what happens when chasis by this ID is not found? Will api client raise exception or will we end up with malfunctioning `RedfishChassis` instance? I'd prefer if we return `None` rather than malfunctioning instance.

Reply:
We are relying on the caller of the method to know what it is using the instance for.
If the id points to no resource, `self.api_client.find` will throw an exception.

Reply:
@matejart what type of exception is raised? I'm thinking it should be caught and `wrapanapi.exceptions.ItemNotFound` raised instead, or a new specific exception type added for `RedfishItemNotFound`.  That way caller has a consistent expectation for what the exception will be if their 'Item' resource_id isn't found, and it doesn't appear to be an exception in the core API.